### PR TITLE
Library to package

### PR DESCRIPTION
**Pull request by dglaude**
This split the single file library into a multi-file package in a folder.
That way I2C version and UART version can be included separately while sharing some code in __init__.

### CODE REVIEW

**Comment by dglaude** [comment]
It seems to pass all the test. This is a fix for #5 that limit memory usage in any case.
Please consider this better than #7

File: examples/pm25_simpletest.py [hunk 1]
```diff
@@ -1,13 +1,15 @@
...
 
 # pylint: disable=unused-import
 import time
 import board
 import busio
 from digitalio import DigitalInOut, Direction, Pull
-import adafruit_pm25
+from adafruit_pm25.i2c import PM25_I2C
+
+# from adafruit_pm25.uart import PM25_UART
```
Comment (brentru) [inline examples/pm25_simpletest.py]: You may want to move this import under the `# Connect to a PM2.5 sensor over UART` or say "uncomment this if you are using UART`

File: adafruit_pm25/uart.py [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+            time.sleep(1)
+
+        self._uart = uart
+        super().__init__()
+
+    def _read_into_buffer(self):
+        while True:
+            b = self._uart.read(1)
+            if not b:
+                print(b)  ### <<< debug
```
Comment (brentru) [inline adafruit_pm25/uart.py]: Don't `print` in the library

**Review by brentru — CHANGES_REQUESTED** [review]
This looks good - thanks for adding an extra protocol - requesting two small nit-picky changes

File: adafruit_pm25/__init__.py [hunk 3]
```diff
@@ -44,10 +44,7 @@
 """
 
 # imports
```
Comment (kattni) [inline adafruit_pm25/__init__.py]: Please delete the `# imports` comment from this line.

File: adafruit_pm25/i2c.py [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pm25`
```
Comment (kattni) [inline adafruit_pm25/i2c.py]: This needs to be:
```suggestion
`adafruit_pm25.i2c`
```

File: adafruit_pm25/i2c.py [hunk 5]
```diff
@@ -0,0 +1,84 @@
...
+================================================================================
+
+CircuitPython library for PM2.5 Air Quality Sensors
```
Comment (kattni) [inline adafruit_pm25/i2c.py]: Please update this line to reflect it being the i2c module for the library, such as:
```suggestion
I2C module for CircuitPython library for PM2.5 Air Quality Sensors
```

File: adafruit_pm25/i2c.py [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+
+
+* Author(s): ladyada
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+Works with most (any?) Plantower UART or I2C interfaced PM2.5 sensor.
```
Comment (kattni) [inline adafruit_pm25/i2c.py]: This should not mention UART as it is now an I2C module.

File: adafruit_pm25/i2c.py [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+# imports
+import time
+from digitalio import Direction
+from adafruit_bus_device.i2c_device import I2CDevice
+from . import PM25
+
+
+class PM25_I2C(PM25):
+    """
+    A driver for the PM2.5 Air quality sensor over I2C
```
Comment (kattni) [inline adafruit_pm25/i2c.py]: Please update to:
```suggestion
    A module for using the PM2.5 Air quality sensor over I2C
```

File: adafruit_pm25/uart.py [hunk 8]
```diff
@@ -0,0 +1,85 @@
...
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+"""
+`adafruit_pm25`
```
Comment (kattni) [inline adafruit_pm25/uart.py]: This should be:
```suggestion
`adafruit_pm25.uart`
```

File: adafruit_pm25/uart.py [hunk 9]
```diff
@@ -0,0 +1,85 @@
...
+================================================================================
+
+CircuitPython library for PM2.5 Air Quality Sensors
```
Comment (kattni) [inline adafruit_pm25/uart.py]: Please update this to reflect being the UART module, such as:
```suggestion
UART module for CircuitPython library for PM2.5 Air Quality Sensors
```

File: adafruit_pm25/uart.py [hunk 10]
```diff
@@ -0,0 +1,85 @@
...
+
+
+* Author(s): ladyada
+
+Implementation Notes
+--------------------
+
+**Hardware:**
+
+Works with most (any?) Plantower UART or I2C interfaced PM2.5 sensor.
```
Comment (kattni) [inline adafruit_pm25/uart.py:36]: Should not mention I2C as this is the UART module.

File: adafruit_pm25/uart.py [hunk 11]
```diff
@@ -0,0 +1,85 @@
...
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
```
Comment (kattni) [inline adafruit_pm25/uart.py]: Is Bus Device necessary for UART?

File: adafruit_pm25/uart.py [hunk 12]
```diff
@@ -0,0 +1,85 @@
...
+
+"""
+
+# imports
```
Comment (kattni) [inline adafruit_pm25/uart.py]: Please remove this comment.

File: adafruit_pm25/i2c.py [hunk 13]
```diff
@@ -0,0 +1,84 @@
...
+
+**Software and Dependencies:**
+
+* Adafruit CircuitPython firmware for the supported boards:
+  https://github.com/adafruit/circuitpython/releases
+* Adafruit's Bus Device library: https://github.com/adafruit/Adafruit_CircuitPython_BusDevice
+
+"""
+
+# imports
```
Comment (kattni) [inline adafruit_pm25/i2c.py:46]: Please remove this comment.

File: examples/pm25_simpletest.py [hunk 14]
```diff
@@ -1,13 +1,15 @@
 """
-Example sketch to connect to PM2.5 sensor with either I2C or UART.
+Example sketch to connect to PM2.5 sensor with I2C.
```
Comment (kattni) [inline examples/pm25_simpletest.py]: It appears you include the UART code commented out. Why did you change the docstring to remove UART?

File: examples/pm25_simpletest.py (hunk 1 above)
Comment (kattni) [reply examples/pm25_simpletest.py]: Please move this under `# Connect to a PM2.5 sensor over UART`. I believe that is a better place for it, and it will be more obvious that it should be uncommented for UART.

**Review by kattni — CHANGES_REQUESTED** [review]
@dglaude This looks really good. There are some necessary changes for Sphinx to build properly, noted below. 

You also need to update this file: https://github.com/adafruit/Adafruit_CircuitPython_PM25/blob/master/docs/api.rst to look more like this for example: https://github.com/adafruit/Adafruit_CircuitPython_CircuitPlayground/blob/main/docs/api.rst

Without updating that, Sphinx will not render it properly. I may have missed something, we'll find out if the checks fail after you make the listed changes.

Thanks!

File: adafruit_pm25/uart.py (hunk 11 above)
Comment (dglaude) [reply adafruit_pm25/uart.py]: Indeed, that what the whole point of this separation.

File: examples/pm25_simpletest.py (hunk 1 above)
Comment (dglaude) [reply examples/pm25_simpletest.py]: Done, but I start to wonder if it would not be best to have two separate "simpletest", one for I2C and one for UART. Just to keep think more simple.

File: adafruit_pm25/uart.py (hunk 2 above)
Comment (dglaude) [reply adafruit_pm25/uart.py]: Done

File: examples/pm25_simpletest.py (hunk 14 above)
Comment (dglaude) [reply examples/pm25_simpletest.py]: restaured

**Comment by dglaude** [comment]
I think I resolved all the change request, including api.rst.
The only thing I would like to argue is that there should be two example rather than one, an I2C version and an UART version.

Should I rename and simplify the current by remove all the UART comment, and make an UART only version?

**Review by brentru — APPROVED** [review]

**Comment by dglaude** [comment]
Maybe those two learn guide will need attention once this is approuved:

- https://learn.adafruit.com/pm25-air-quality-sensor/python-and-circuitpython
- https://learn.adafruit.com/pmsa003i/python-circuitpython

**Comment by FoamyGuy** [comment]
I found another learn guide that will also need a small update when this PR gets merged. https://learn.adafruit.com/diy-air-quality-monitor/code-usage

@kattni and @brentru If this is ready to merge, I can work on updating the learn guides and their code if you want?

**Comment by kattni** [comment]
@FoamyGuy Please merge this and release the library, and then proceed onto updating the guides. That would be greatly appreciated - thank you!
